src/packagedcode/models.py

```diff
@@ -1130,6 +1136,33 @@ def create_default_package_data(cls, **kwargs):
             **kwargs,
         )
 
+    @classmethod
+    def populate_license_fields(cls, package_data):
+        """
+        Run license detection on extracted_license_statement for a PackageData
+        object, and add the declared_license_expression (and the spdx expression)
+        and corresponding LicenseDetection data.
+        """
+        if not package_data.declared_license_expression and package_data.extracted_license_statement:
```

Comment:
What about inverting and splitting the ifs for readability?
```suggestion
        if package_data.declared_license_expression:
            return
        if not package_data.extracted_license_statement:
            return
```

Reply:
Yes, added this.